### PR TITLE
chore: roll to Playwright v1.45.0

### PR DESCRIPTION
**Pull request by canstand**

### CODE REVIEW

File: browser.go [hunk 1]
```diff
@@ -84,6 +84,20 @@ func (b *browserImpl) NewContext(options ...BrowserNewContextOptions) (BrowserCo
 	context := fromChannel(channel).(*browserContextImpl)
 	context.browser = b
 	b.browserType.(*browserTypeImpl).didCreateContext(context, &option, nil)
+	if os.Getenv("PW_FREEZE_TIME") != "" {
```
Comment (mxschmitt) [inline browser.go:87]: I'd remove this, since we are testing this kind of stuff only in the microsoft/playwright repository.

**Review by mxschmitt — APPROVED** [review]

File: browser.go (hunk 1 above)
Comment (canstand) [reply browser.go]: It's been removed.  And I need to ask you to merge this PR, which I can't due to `macos-13` in CI is switched back to `macos-latest`.

**Comment by canstand** [comment]
Hi @mxschmitt  [firefox test timeout](https://github.com/microsoft/playwright/issues/30705) regression in playwright v1.45.1

**Comment by mxschmitt** [comment]
> firefox test timeout

This is not fixed yet, only in Firefox Beta, which will be normal Firefox in v1.46.0. So I think we need to revert your workflow change until then.

**Pull request merged**